geode-management/src/main/java/org/apache/geode/management/api/ClusterManagementException.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management.api;
+
+/**
+ * Base class of all exceptions thrown by {@link ClusterManagementService} implementations.
+ */
+public class ClusterManagementException extends RuntimeException {
+  protected final ClusterManagementResult result;
+
+  public ClusterManagementException(ClusterManagementResult result) {
+    super(result.getStatusCode()
+        + (org.springframework.util.StringUtils.isEmpty(result.getStatusMessage()) ? ""
+            : (": " + result.getStatusMessage())));
+    this.result = result;
+  }
+
+  public ClusterManagementResult getResult() {
+    return result;
+  }
+
+  public ClusterManagementResult.StatusCode getStatusCode() {
```

Comment:
consider removing getStatusCode() and getStatusMessage() since they can be done using getResult().getStatusCode() ...

Reply:
I'd prefer the other direction, since getResult will have a lot of fields that are not applicable.  I would remove `getResult()` entirely but the controller advice needs it.  Marked as internal use only.